network/tcp.go

```diff
@@ -58,7 +58,10 @@ func NewTCPConn(addr Address) (*TCPConn, error) {
 		}
 		time.Sleep(WaitRetry)
 	}
-	return nil, fmt.Errorf("Could not connect to %s: %s", addr, err)
+	if err != nil {
+		return nil, fmt.Errorf("Could not connect to %s: %s", addr, err)
+	}
+	return nil, fmt.Errorf("Timeout while connecting to %s", addr)
```

Comment:
Added `ErrTimeout` and correct waiting (don't wait if it failed on the last count).